icicle-compiler/test/Icicle/Test/Arbitrary/Data.hs

```diff
@@ -49,6 +53,22 @@ instance IsString Var where
 fresh :: Int -> Name Var
 fresh = nameOf . NameBase . Var "_fresh"
 
+-- Sometimes it's nice to be able to pretty print our generated programs
+instance PP.Pretty Var where
+ pretty (Var t i) = PP.text (T.unpack t) <> PP.text (show i)
+
+-- Generate totally arbitrary, totally random variables
+instance Arbitrary Var where
+  arbitrary =
+   sized $ \size ->
+    Var <$> elements viruses <*> choose (0, size)
```

Comment:
please use `cats`? `viruses` have something with a space in it, very annoying to read :(